classes/form/CustomerForm.php

```diff
@@ -127,6 +127,13 @@ public function validate()
             }
         }
 
+        $passwordField = $this->getField('password');
+        if (!Validate::isPasswd($passwordField->getValue())) {
```

Comment:
You should probably check the value is not empty at the same time.

```suggestion
        $passwordValue = $passwordField->getValue();
        if (!empty($passwordValue) && !Validate::isPasswd($passwordValue)) {
```

Reply:
And it maybe should be include in `validateFieldsLengths` instead.

Reply:
Why validateFieldsLenghts? The password validation can be extended for accepting only string with at least "one capital letter, one lowercase letter, one number and one symbol".
The password field validation should be an atomic function. 
I'm not sure that allowing empty password for guest in the same point could a good practice.
Maybe the code should check:
- if the password is mandatory (no guest allowed) OR inserted.
-- TRUE: password have to be valid
-- FALSE: if password is mandatory -> ERROR. Else OK